### PR TITLE
v0.3.0 - Deprecated Extension

### PR DESCRIPTION
**Pull request by gadenbuie**
This replaces the `Posit.shiny-python` extension with a shell extension that

1. Directly depends on the new `Posit.shiny` extension to ensure updating users get the new extension.
2. Checks that the new extension is installed on activation in case somebody slips through the cracks.
3. Points to the new extension in the README and CHANGELOG

For #53

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,56 +1,5 @@
...
-
-### Shinylive
-
-- `shiny.shinylive.appMode`: Should the Shinylive link open the app in `"app"` mode, showing only the app and an optional header, or in `"editor"` mode with the app alongside an editor and console pane. The default is `"ask"`, which prompts you each time you create Shinylive link.
-- `shiny.includeHeader`: Include the "Shiny" header in the Shinylive link when opening in app mode?
-- `shiny.shinylive.openAction`: What action should be taken when opening a Shinylive link? Options are `"open"` to open the link in an external browser, `"copy"` the link to the clipboard, or `"ask"`. The default is `"ask"`, which prompts you each time you create a Shinylive link.
-- `shiny.shinylive.host`: The Shinylive host used when creating a Shinylive link. The default is `"https://shinylive.io"`, which uses the latest released version of Shiny in Python or R. Or `"https://posit-dev.github.io/shinylive"`, which uses the latest development version of Shiny in Python or R.
+New Shiny users should install the [Shiny extension](https://marketplace.visualstudio.com/items?itemName=Posit.shiny) instead of this one.
```
Comment (jcheng5) [inline README.md:5]: Include links for both VS Code Marketplace and Open-VSX.

File: src/extension.ts [hunk 2]
```diff
@@ -1,180 +1,38 @@
...
-    ),
-    vscode.commands.registerCommand(
-      "shiny.shinylive.saveAppFromUrl",
-      shinyliveSaveAppFromUrl
-    ),
-    vscode.commands.registerCommand(
-      "shiny.shinylive.createFromExplorer",
-      shinyliveCreateFromExplorer
-    )
+export async function activate(
```
Comment (jcheng5) [inline src/extension.ts:3]: Either don't do this at all, or, unconditionally offer to uninstall yourself. (And remove that offer from `Posit.shiny`)

Comment (jcheng5) [reply src/extension.ts:3]: Would it be crazy to unconditionally uninstall, without even asking?

Comment (gadenbuie) [reply src/extension.ts]: I went with the unconditional uninstall. I thought about including an informational message telling the user about the action but decided against it. Let me know if you'd like to add that back in.

File: src/extension.ts [hunk 3]
```diff
@@ -1,38 +1,17 @@
...
-  // This shell `Posit.shiny-python` extension depends on the new `Posit.shiny`
-  // extension. In the unlikely event the new extension is not installed, this
-  // check will prompt the user to install it. The new extension handles
-  // uninstallation or deactivation of the `Posit.shiny-python` extension.
-  if (vscode.extensions.getExtension("Posit.shiny")) {
-    return;
+export async function activate(): Promise<void> {
+  // If the Posit.shiny extension is installed, this old shell extension is no
+  // longer needed.
+  if (!vscode.extensions.getExtension("Posit.shiny")) {
```
Comment (jcheng5) [inline src/extension.ts:6]: Since we have a dependency on Posit.shiny, is it even possible to get into this state?

**Review by jcheng5 — APPROVED** [review]

File: src/extension.ts (hunk 3 above)
Comment (gadenbuie) [reply src/extension.ts:6]: I don't think so? Or at least it's very unlikely. So this is purely defensive.

**Pull request closed without merging**